core/execution/future/market.go

```diff
@@ -4374,6 +4381,9 @@ func (m *Market) settlementDataPerp(ctx context.Context, settlementData *num.Num
 	if sdi == nil {
 		return
 	}
+
+	m.underlyingIndexPrice = sdi
```

Comment:
Here `sdi` is the funding-payment at the end of the period, and not the last external price received from the data-sources. If you want the last external price, you just need to grab:
```
externalPrice = p.externalTWAP.points[-1]
```
somewhere here:
https://github.com/vegaprotocol/vega/blob/develop/core/products/perpetual.go#L751